userspace/libscap/scap.c

```diff
@@ -1513,6 +1513,51 @@ uint64_t scap_get_driver_schema_version(scap_t* handle)
 	return 0;
 }
 
+void scap_gethostname(scap_t* handle)
+{
+	gethostname(handle->m_machine_info.hostname, sizeof(handle->m_machine_info.hostname) / sizeof(handle->m_machine_info.hostname[0]));
+	char *env_hostname = getenv(SCAP_HOSTNAME_ENV_VAR);
+	if(env_hostname != NULL)
+	{
+		snprintf(handle->m_machine_info.hostname, sizeof(handle->m_machine_info.hostname), "%s", env_hostname);
+	}
+}
+
+void scap_get_self_pid_ts_epoch(scap_t* handle)
+{
+	handle->m_machine_info.self_pid_ts_epoch = 0;
+#ifdef __linux__
+	struct stat st = {0};
+	char path[256];
+	// proc dir as seen from the calling process pid namespace
+	snprintf(path, sizeof(path), "/proc/%d/", getpid());
+	if(stat(path, &st) == 0)
+	{
+		handle->m_machine_info.self_pid_ts_epoch = st.st_ctim.tv_sec * (uint64_t) SECOND_TO_NS + st.st_ctim.tv_nsec;
+	}
+#endif
+}
+
+void scap_get_bpf_stats_enabled(scap_t* handle)
+{
+	FILE* f;
+	// proc dir as seen from the calling process pid namespace
```

Comment:
```suggestion
	handle->m_machine_info.flags &= ~PPM_BPF_STATS_ENABLED;
	// proc dir as seen from the calling process pid namespace
```
Making sure we clear out the flag bit.